src/Val.h

```diff
@@ -89,67 +89,21 @@ using TableValPtr = IntrusivePtr<TableVal>;
 using ValPtr = IntrusivePtr<Val>;
 using VectorValPtr = IntrusivePtr<VectorVal>;
 
-union BroValUnion {
-	// Used for bool, int, enum.
-	bro_int_t int_val;
-
-	// Used for count, counter, port.
-	bro_uint_t uint_val;
-
-	// Used for addr
-	IPAddr* addr_val;
-
-	// Used for subnet
-	IPPrefix* subnet_val;
-
-	// Used for double, time, interval.
-	double double_val;
-
-	String* string_val;
-	Func* func_val;
-	File* file_val;
-	RE_Matcher* re_val;
-	PDict<TableEntryVal>* table_val;
-	std::vector<ValPtr>* record_val;
-	std::vector<ValPtr>* vector_val;
-
-	BroValUnion() = default;
-
-	constexpr BroValUnion(bro_int_t value) noexcept
-		: int_val(value) {}
-
-	constexpr BroValUnion(bro_uint_t value) noexcept
-		: uint_val(value) {}
-
-	constexpr BroValUnion(IPAddr* value) noexcept
-		: addr_val(value) {}
-
-	constexpr BroValUnion(IPPrefix* value) noexcept
-		: subnet_val(value) {}
-
-	constexpr BroValUnion(double value) noexcept
-		: double_val(value) {}
-
-	constexpr BroValUnion(String* value) noexcept
-		: string_val(value) {}
-
-	constexpr BroValUnion(Func* value) noexcept
-		: func_val(value) {}
-
-	constexpr BroValUnion(File* value) noexcept
-		: file_val(value) {}
-
-	constexpr BroValUnion(RE_Matcher* value) noexcept
-		: re_val(value) {}
-
-	constexpr BroValUnion(PDict<TableEntryVal>* value) noexcept
-		: table_val(value) {}
-};
-
 class Val : public Obj {
 public:
 	static inline const ValPtr nil;
 
+#ifdef DEPRECATED
```

Comment:
One other thought: we could specialize `make_intrusive<Val>(...)` so that depending on the types of the its argument, it creates the right derived class. Not sure how much that would help, given that `make_intrusive` is itself pretty new.

Reply:
Neat thought on refining `make_intrusive<Val>` - but I suspect you're right, if there's code using those constructors much of it will be doing so via `new`.

Reply:
Yeah, probably not worth it. I keep my suggestion on the ValManager, but we can postpone that for later, given that it'd be a new API, nothing we're changing here.

Reply:
Okay, for now I'll go ahead and remove the `#ifdef` block and leave it at that.